### PR TITLE
Enable Algolia search with light theme only

### LINKED ISSUE #42: Enable search
- We have more content and it is hard to find if the website has a content that we are looking for.
- We should enable search feature to our website.

* [x] Try with built-in Gatsby search index at build time
* [x] The try with a third party service
* [ ] Try with Google search bar.

### PR DESCRIPTION
**Pull request by mildronize**
Please review my PR, I will send the environment secret later

```
ALGOLIA_APP_ID=xxxxx
ALGOLIA_SEARCH_KEY=xxxxx
ALGOLIA_ADMIN_KEY=xxxxx
```

### CODE REVIEW

File: src/components/Search/algolia/index.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ import styled from '@emotion/styled';
 import SearchStatus from '../Status';
 import Pagination from './pagination';
 import Stats from './stats';
+import AlgoliaLogo from './AlgoliaLogo';
```
Comment (aaronamm) [inline src/components/Search/algolia/index.js:18]: So to use a free plan/open source project, we need to put their logo in our website right?

File: src/components/Search/Input.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ const Input = styled.input`
   border-width: 0 !important;
   &,
   ::placeholder {
-    color: ${(props) => props.theme.colors.gray};
+    color: ${(props) => props.theme.colors.font};
```
Comment (aaronamm) [inline src/components/Search/Input.js:46]: Why do we need to change a font color?

File: .github/workflows/publish-to-github-pages.yml [hunk 3]
```diff
@@ -9,11 +9,15 @@ on:
...
+      ALGOLIA_ADMIN_KEY: ${{ secrets.ALGOLIA_ADMIN_KEY }}
     steps:
       - uses: actions/checkout@master
 
       - uses: enriikke/gatsby-gh-pages-action@v2
         with:
           # access-token is a GitHub Personal access token with the "repo" scope
           access-token: ${{ secrets.PUBLIC_REPO_ACCESS_TOKEN }}
-          deploy-branch: gh-pages
+          deploy-branch: gh-pages
```
Comment (aaronamm) [inline .github/workflows/publish-to-github-pages.yml:23]: Very minor, we prefer to have empty line at the end of file.

File: config/default.js [hunk 4]
```diff
@@ -61,11 +61,11 @@ module.exports = {
     search: {
       enabled: true,
       indexName: 'docs',
-      algoliaAppId: null,
-      algoliaSearchKey: null,
-      algoliaAdminKey: null,
-      excerptSize: 100000,
-      engine: 'localsearch',
+      algoliaAppId: process.env.ALGOLIA_APP_ID,
```
Comment (aaronamm) [inline config/default.js:64]: What your configure in GitHub Actions workflow script is correct.
However, I don't understand how our project builds indexes.

Will indexes get built when calling `yarn build` in package.json which is called by GitHub Actions
https://github.com/enriikke/gatsby-gh-pages-action/blob/main/index.ts#L44?
Do we need extra command for building indexest?

**Review by aaronamm — CHANGES_REQUESTED** [review]

File: src/components/Search/algolia/index.js (hunk 1 above)
Comment (mildronize) [reply src/components/Search/algolia/index.js:18]: Yes

File: src/components/Search/Input.js (hunk 2 above)
Comment (mildronize) [reply src/components/Search/Input.js:46]: it's broken Dark mode color

File: .github/workflows/publish-to-github-pages.yml (hunk 3 above)
Comment (mildronize) [reply .github/workflows/publish-to-github-pages.yml:23]: Sorry

File: config/default.js (hunk 4 above)
Comment (mildronize) [reply config/default.js:64]: It will be done automatically by `gatsby-plugin-algolia`, when run `yarn build`, it'll build index.

Comment (aaronamm) [reply config/default.js:64]: Thanks.

**Review by aaronamm — APPROVED** [review]

**Pull request merged**